src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1028,8 +1046,42 @@ protected function prepare_items_query( $prepared_args = array(), $request = nul
 				'include_slugs' => 'post_name__in',
 			);
 
-			if ( isset( $orderby_mappings[ $request['orderby'] ] ) ) {
-				$query_args['orderby'] = $orderby_mappings[ $request['orderby'] ];
+			// The orderby query string parameter can look like ?orderby=menu_order, ?orderby[]=menu_order, or ?orderby[menu_order]=asc.
+			if ( rest_is_array( $request['orderby'] ) ) {
+				// An array: ?orderby[]=menu_order
+				// A fake array: ?orderby=menu_order,title
+				// A string: ?orderby=menu_order
+				// The latter two because rest_is_array() and rest_sanitize_array() pass scalars through wp_parse_list()
+				$orderby_mappings_in_request = array_intersect_key( $orderby_mappings, array_flip( $request['orderby'] ) );
+
+				$new_arg = array();
+				// Remaps values. Preserves value order.
+				foreach ( $query_args['orderby'] as $orderby ) {
+					if ( array_key_exists( $orderby, $orderby_mappings_in_request ) ) {
+							$new_arg[] = $orderby_mappings_in_request[ $orderby ];
```

Comment:
Looks like I broke some whitespace here, sorry!

```suggestion
						$new_arg[] = $orderby_mappings_in_request[ $orderby ];
```

Reply:
Good eye, that might have been from me